storage/src/test/java/zipkin2/storage/kafka/KafkaStorageIT.java

```diff
@@ -167,4 +167,302 @@ public void shouldFindTraces() throws Exception {
           return traces.size() == 1 && traces.get(0).size() == 2;
         });
   }
-}
\ No newline at end of file
+
+  @Test
+  public void shouldFindTracesByAnnotation() throws IOException, InterruptedException {
+
+    // todo: it is not explicit that annotations applied for tags
+    Map<String, String> annotationQuery =
+        new HashMap<String, String>() {
+          {
+            put("key_tag_a", "value_tag_a");
+          }
+        };
+
+    Span span1 =
+        Span.newBuilder()
+            .traceId("a")
+            .id("a")
+            .putTag("key_tag_a", "value_tag_a")
+            .localEndpoint(Endpoint.newBuilder().serviceName("svc_a").build())
+            .name("op_a")
+            .kind(Span.Kind.CLIENT)
+            .timestamp(Long.valueOf(TODAY + "000"))
+            .duration(10)
+            .build();
+
+    Span span2 =
+        Span.newBuilder()
+            .traceId("b")
+            .id("b")
+            .localEndpoint(Endpoint.newBuilder().serviceName("svc_b").build())
+            .putTag("key_tag_c", "value_tag_d")
+            .addAnnotation(Long.valueOf(TODAY + "000"), "annotation_b")
+            .name("op_b")
+            .kind(Span.Kind.CLIENT)
+            .timestamp(Long.valueOf(TODAY + "000"))
+            .duration(10)
+            .build();
+
+    List<Span> spans = Arrays.asList(span1, span2);
+    storage.spanConsumer().accept(spans).execute();
+
+    IntegrationTestUtils.waitUntilMinRecordsReceived(
+        testConsumerConfig, storage.spansTopic.name, 2, 10000);
+
+    // query by annotation {"key_tag_a":"value_tag_a"} = 1 trace
+    await()
+        .pollDelay(5, TimeUnit.SECONDS)
+        .until(
+            () -> {
+              List<List<Span>> traces =
+                  storage
+                      .spanStore()
+                      .getTraces(
+                          QueryRequest.newBuilder()
+                              .annotationQuery(annotationQuery)
+                              .endTs(TODAY + 1)
+                              .limit(10)
+                              .lookback(Duration.ofMinutes(1).toMillis())
+                              .build())
+                      .execute();
+              return traces.size() == 1;
+            });
+
+    // query by annotation {"key_tag_non_exist_a":"value_tag_non_exist_a"} = 1 trace
+    await()
+        .pollDelay(5, TimeUnit.SECONDS)
```

Comment:
Same here.

Reply:
There are some cases, where it should not found traces (f.ex. with non_existing_service_name), in that case exactly need delayPoll.
I will update comment